lib/jobs/dell-wsman-simple-update-firmware.js

```diff
@@ -0,0 +1,249 @@
+    configuration,
+    fs,
+    ChildProcess
+) {
+    var logger = Logger.initialize(WsmanSimpleFirmwareUpdateJobFactory);
+
+    /*
+     * @param {object} options
+     * @param {object} context
+     * @param {string} taskId
+     * @constructor
+     */
+    function WsmanSimpleFirmwareUpdateJob(options, context, taskId) {
+        var self = this;
+        WsmanSimpleFirmwareUpdateJob.super_.call(self, logger, options, context, taskId);
+        assert.object(self.options);
+        logger.debug('Taskgraph options: ' + JSON.stringify(self.options));
+        self.graphId = self.context.graphId;
+        self.imageURI = self.options.imageURI;
+        self.imageFileName = self.imageURI.slice(self.imageURI.lastIndexOf('/') + 1);
+        self.path = self.imageURI.slice(0, self.imageURI.lastIndexOf('/') + 1);
+        self.componentIds = [];
+        self.targetConfig = {};
+    }
+
+    util.inherits(WsmanSimpleFirmwareUpdateJob, BaseJob);
+
+    /*
+     * Initialize basic configuraion for the job
+     */
+    WsmanSimpleFirmwareUpdateJob.prototype._initJob = function() {
+        var self = this;
+        return Promise.try(function(){
+            var dellConfig = configuration.get('dell');
+            if(!dellConfig || !dellConfig.gateway || !dellConfig.services.firmware) {
+                throw new errors.NotFoundError('Dell firmware update SMI service is not defined in smiConfig.json.');
+            }
+            self.firmwareConfigs = dellConfig.services.firmware;
+            self.apiServer = dellConfig.gateway;
+            if(!self.firmwareConfigs.dupUpdater) {
+                throw new errors.NotFoundError('Firmware dup updater is not defined in Smiconfig.json.');
+            }
+        })
+        .then(function() {
+            return self.getComponentType();
+        })
+        .then(function() {
+            return self.checkOBM('Firmware Update');
+        })
+        .then(function(obm) {
+            if(!(obm.config && obm.config.host && obm.config.user && obm.config.password)) {
+                throw new errors.NotFoundError('Dell obm setting is invalid.');
+            } else {
+                self.targetConfig.serverAddress = obm.config.host;
+                self.targetConfig.username = obm.config.user;
+                self.targetConfig.password = obm.config.password;
+            }
+        })
+        .catch(function(err) {
+            logger.error('An error occurred while initializing job.', { error: err });
+            throw err;
+        });
+    };
+
+    /*
+     * Create tmp folder to store downloaded image file. (/tmp/graphid)
+     * Download image file into graphid folder.
+     * Unzip package.xml from image file. Get component id(s) for sending firmware update request.
+     */
+    WsmanSimpleFirmwareUpdateJob.prototype.getComponentType = function() {
+        var self = this;
+        var graphTempFolder = '/tmp/' + self.graphId;
+        var destinationFile = '/tmp/' + self.imageFileName;
```

Comment:
This API is better for creating the temp directory: https://nodejs.org/api/fs.html#fs_fs_mkdtemp_prefix_options_callback

Reply:
Yes, it seems FS API have good encapsulation. But as team discussed before, name the temp folder with graph id also make sense, not sure if could just keep it that way. @yyscamper